Support JIRA links with domain paths (reverse proxy setups)

danriedl opened the pull request:
## Summary
This PR adds support for JIRA links that include domain paths, such as those used in reverse proxy setups (e.g. mycompany.net/jira/browse/ISSUE-123). Previously, the plugin only recognized links where the /browse/ segment directly followed the domain, which did not work for setups where JIRA is hosted behind a path (e.g. /jira/).

### Motivation
Many enterprise environments deploy JIRA behind a reverse proxy or under a sub-path, resulting in URLs like https://mycompany.net/jira/browse/ISSUE-123. The previous implementation did not recognize these links, so they were not shortened as expected.

### Changes
- Regex updated to support arbitrary paths between the domain and /browse/
- Tests added and unified for reverse proxy and path support
- Setting description updated to clarify that paths are supported

### Impact
The plugin now works for both standard JIRA cloud/server URLs and custom setups with domain paths. All existing functionality is preserved and extended.

## Test plan
- All unit tests pass
- Manual verification with both standard and reverse proxy JIRA URLs

danriedl committed:
doc: update setting description to clarify path support

danriedl committed:
feat: support JIRA links with domain paths (reverse proxy setups)

danriedl committed:
ut: add and unify tests for reverse proxy and path support

danriedl committed:
Merge branch 'master' into fix/jira-url-path-support

rplatonovs commented on utils.ts:
```diff
@@ -1,14 +1,15 @@
-	const jiraRegex = new RegExp(`^https?:\/\/${escDomain}\/browse\/([A-Z]+-\\d+)(\\?.*)?`, 'g');
+
+	const escDomain = domain.replace(/\./g, '\\.').replace(/(\*)/g, '.*');
+	const jiraRegex = new RegExp(`^https?:\/\/${escDomain}(\/[\\w\\-\\.]+)*\/browse\/(?<issueId>[A-Z]+-\\d+)(\\?.*)?`, 'g');
```
No need for so many changes in this part, just putting `.*` in front of `browse` should be enough:
```
^https?:\/\/${escDomain}\/.*browse\/([A-Z]+-\\d+)(\\?.*)?
```

danriedl replied:
Thank you for the suggestion! I’ve updated the regex to use .* before /browse as you recommended. This simplifies the pattern and still matches all required JIRA link formats, including those with reverse proxy paths. 
I confirmed, that all tests pass with this change. 👍

rplatonovs commented on utils.ts:
```diff
@@ -1,14 +1,15 @@
 	if (!jiraRegex.test(pastedText)) return null;
 
-	return pastedText.replace(jiraRegex, (match, issueId) => {
+	return pastedText.replace(jiraRegex, (match, ...groups) => {
```
This change becomes unnecessary after tweaking the regex

danriedl replied:
With the simplified regex, the replacement function can use the original argument structure. 
I’ve reverted this part accordingly.

danriedl committed:
fix(utils): simplify JIRA link regex and revert replacement function

- Simplifies the JIRA link matching regex to use .* before /browse as suggested by reviewer
- Reverts replacement function to use original argument structure
- No change in functionality; all tests pass
